flaky test: TestBatchMetricsProcessor_Timeout

```
=== RUN   TestBatchMetricsProcessor_Timeout
    assertion_compare.go:342: 
        	Error Trace:	batch_processor_test.go:424
        	Error:      	"100000000" is not less than or equal to "606204"
        	Test:       	TestBatchMetricsProcessor_Timeout
        	Messages:   	[]
--- FAIL: TestBatchMetricsProcessor_Timeout (0.00s)
```

_Originally posted by @codeboten in https://github.com/open-telemetry/opentelemetry-collector/pull/4305#pullrequestreview-793316046_

--- a/processor/batchprocessor/batch_processor_test.go
+++ b/processor/batchprocessor/batch_processor_test.go
@@ -395,7 +395,7 @@ func TestBatchMetricsProcessor_Timeout(t *testing.T) {
 	cfg := Config{
 		ProcessorSettings: config.NewProcessorSettings(config.NewComponentID(typeStr)),
 		Timeout:           100 * time.Millisecond,
-		SendBatchSize:     100,
+		SendBatchSize:     101,
 	}
 	requestCount := 5
 	metricsPerRequest := 10

